### PR TITLE
 chore(ios): prevent modal windows from being swiped down

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-27169

### CODE REVIEW

**Comment by janvennemann** [comment]
@vijaysingh-axway the commits don't appear to be related to the ticket 😄

File: iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m [hunk 1]
```diff
@@ -558,6 +558,13 @@ - (void)openOnUIThread:(NSArray *)args
           [theController setModalPresentationStyle:style];
         }
       }
+
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+      if ([TiUtils isIOSVersionOrGreater:@"13.0"]) {
+        isForceModalDisable = ![TiUtils boolValue:@"forceModal" properties:dict def:NO];
```
Comment (hansemannn) [inline iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m]: So it's creation only?

Comment (vijaysingh-axway) [reply iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m]: This should be pass as parameter while opening window.

Comment (janvennemann) [inline iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m]: The double negation here and in the following line is confusing. Consider changing the variable name so you don't have to negate the result twice, for example:

```suggestion
        forceModalBehavior = [TiUtils boolValue:@"forceModal" properties:dict def:NO];
```

**Review by janvennemann — CHANGES_REQUESTED** [review]

File: iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m [hunk 2]
```diff
@@ -558,6 +558,13 @@ - (void)openOnUIThread:(NSArray *)args
           [theController setModalPresentationStyle:style];
         }
       }
+
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
```
Comment (janvennemann) [inline iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m]: Use the new `IS_IOS_SDK_13` constant

Comment (vijaysingh-axway) [reply iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m]: This macro will not work inside framework.

Comment (janvennemann) [reply iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m]: Duh, of course.

File: iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m (hunk 1 above)
Comment (vijaysingh-axway) [reply iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m]: Updated.

**Comment by hansemannn** [comment]
@vijaysingh-axway Can you squash the commits please? Otherwise it's hard to cherry pick

**Comment by hansemannn** [comment]
We tested this in two places of our app and it does not work. Did you try this in an Alloy based app (inside the tss)?

File: iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m [hunk 3]
```diff
@@ -46,6 +46,7 @@ - (void)_destroy
 
 - (void)_configure
 {
+  forceModal = YES;
```
Comment (janvennemann) [inline iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m]: Why is this set to `YES` here when it's default value is defined as `NO`?

Comment (janvennemann) [reply iphone/TitaniumKit/TitaniumKit/Sources/API/TiWindowProxy.m]: Ah, I was just wondering what's the reason behind that. Thanks for the explanation.

**Review by janvennemann — APPROVED** [review]

**Review by ssjsamir — APPROVED** [review]
**FR Passed** Modal windows no longer being swiped down. Tested with the two test case from https://jira.appcelerator.org/browse/TIMOB-27169

**Test Environment**
```
MacOS Mojave version 10.14.4
Xcode 11 beta 5
Node.js ^8.11.1
iPhone 8 sim (13.0)
"NPM":"4.2.14","CLI":"7.1.1-master.2"
```

**Pull request merged**